src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala

```diff
@@ -134,13 +136,16 @@ object DockerPlugin extends AutoPlugin {
       val gidOpt = (daemonGroupGid in Docker).value
       val base = dockerBaseImage.value
       val addPerms = dockerAdditionalPermissions.value
+      val uniqueDockerfileId = UUID.randomUUID().toString
 
       val generalCommands = makeFrom(base) +: makeMaintainer((maintainer in Docker).value).toSeq
       val stage0name = "stage0"
       val stage0: Seq[CmdLike] = strategy match {
         case DockerPermissionStrategy.MultiStage =>
           Seq(
             makeFromAs(base, stage0name),
+            makeLabel("sbt-native-packager-multi-stage" -> "intermediate"),
+            makeLabel("sbt-native-packager-multi-stage-id" -> uniqueDockerfileId),
```

Comment:
Can you explain what issue we solve with two labels? From my limited understand we label an image so we can prune it afterwards with `--filter label=our-label`.

From my naive understand the `intermediate` label should be sufficient? Granted that some uniqueness is required so we don't accidentally prune other images. However something with `normalizedName in Docker` would be probably be enough (e.g. `val uniqueDockerfileId = (normalizedName in Docker).value + "-" + (version in Docker).value + "-intermediate".

Reply:
> Can you explain what issue we solve with two labels?

One is a very specific label, exactly for one specific build. My intention was that at the end of the build only exactly the intermediate image(s) of that build should be removed, without any possible side effects.
The general label was just added as help in case you disable the autoremove config and want to clean all images later, or in case something went wrong (and cleaning did not happen correctly) and you want to clean manually.

> From my naive understand the `intermediate` label should be sufficient? 

For the mentioned intention above, not it's not. Because we would potentially clean images that the user does not want to clean up. Who knows, maybe a user kept an intermediate image on purpose to inspect it, and when running the build again, suddenly that image would be gone.

> Granted that some uniqueness is required so we don't accidentally prune other images. 

Exactly, that's my whole point. My intention was to not have any side-effects at all.

> However something with `normalizedName in Docker` would be probably be enough (e.g. `val uniqueDockerfileId = (normalizedName in Docker).value + "-" + (version in Docker).value + "-intermediate".

I am not sure this is enough. Sometimes when setting up a build process (like we did recently via CI/CD), we just package dozens of docker packages until we have a correct build process set up. Now, these package all have the same name, (snapshot) version, etc. Now again my argument: Maybe a user wants to keep an intermediate image on purpose, however if you clean based on name + version you still would clean images a user wanted to keep.